server/bleep/src/webserver/answer.rs

```diff
@@ -503,16 +527,48 @@ impl Conversation {
             }
         }
 
-        let messages = self
-            .llm_history
-            .iter()
-            .filter(|m| m.role == "user")
-            .map(|m| &m.content)
-            .collect::<Vec<_>>();
+        let context = {
+            let mut s =
+                "Below is the current context, Future actions will add to this.\n".to_owned();
+
+            if !self.path_aliases.is_empty() {
+                s += "##### PATHS #####\npath alias, path\n";
+
+                for (alias, path) in self.path_aliases.iter().enumerate() {
+                    s += &format!("{alias}, {path}\n");
+                }
+            }
+
+            let mut has_chunk = false;
+            // Order chunks by most recent.
+            for chunk in self.code_chunks.iter().rev() {
+                if !has_chunk {
+                    has_chunk = true;
+                    s += "\n##### CODE CHUNKS #####\n\n";
+                }
+
+                let snippet = chunk
+                    .snippet
+                    .lines()
+                    .enumerate()
+                    .map(|(i, line)| format!("{}: {line}\n", i + chunk.start_line as usize))
+                    .collect::<String>();
+
+                s += &format!("### path alias: {} ###\n{snippet}\n\n", chunk.alias);
+            }
+
+            s
+        };
+
+        let query = self
+            .exchanges
+            .last()
+            .context("the exchange list was empty")?
```

Comment:
for a new conversation, the exchange list will always be empty, no?

Reply:
Good catch, this is fixed now